cmd/frontend/internal/auth/userpasswd/handlers.go

```diff
@@ -94,6 +115,22 @@ func handleSignUp(w http.ResponseWriter, r *http.Request, failIfNewUserIsNotInit
 		}
 		newUserData.EmailVerificationCode = code
 	}
+
+	// Prevent abuse (users adding emails of other people whom they want to annoy) with the
+	// following abuse prevention checks.
+	if conf.EmailVerificationRequired() && !newUserData.EmailIsVerified {
```

Comment:
Is `conf.EmailVerificationRequired` a necessary condition here or is `checkEmailAbuse` responsible for checking that? (it seems like it does check that already)

```suggestion
	if !newUserData.EmailIsVerified {
```

Reply:
> Is `conf.EmailVerificationRequired` a necessary condition here or is `checkEmailAbuse` responsible for checking that? (it seems like it does check that already)

It is necessary here. The same `if conds` though checks again ~30 lines below for sending email, but that is after the user created, which is too late (doesn't prevent adding a new user to the system).